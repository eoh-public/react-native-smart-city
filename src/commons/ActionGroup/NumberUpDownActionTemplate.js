import React, { useCallback, useEffect, useState } from 'react';
import { View, TouchableOpacity } from 'react-native';
import { IconOutline } from '@ant-design/icons-react-native';
import { Colors } from '../../configs';

import Text from '../Text';
import { useConfigGlobalState } from '../../iot/states';
import styles from './NumberUpDownActionTemplateStyle';
import { watchMultiConfigs } from '../../iot/Monitor';

const NumberUpDownActionTemplate = ({ actionGroup, doAction, sensor }) => {
  const { configuration } = actionGroup;
  const {
    action_data,
    max_value,
    min_value,
    text_format,
    keep_track_config,
    config,
  } = configuration;

  // eslint-disable-next-line no-unused-vars
  const [configValues, setConfigValues] = useConfigGlobalState('configValues');
  const [value, setValue] = useState((config && configValues[config]) || 28);

  useEffect(() => {
    if (!keep_track_config) {
      return;
    }
    if (!config) {
      return;
    }

    const configValue = configValues[config];
    if (configValue !== null && configValue !== undefined) {
      setValue(configValue);
    }
  }, [configValues, config, keep_track_config]);

  const doActionAndWatchConfig = useCallback(
    (actionData, actionValue) => {
      doAction(actionData, actionValue);
      if (!sensor.is_managed_by_backend) {
        return;
      }
      if (!keep_track_config) {
        return;
      }
      if (!config) {
        return;
      }

      config && watchMultiConfigs([config]);
    },
    [config, doAction, keep_track_config, sensor.is_managed_by_backend]
  );

  const doActionUp = useCallback(async () => {
    const newValue = checkMinMax(value + 1);
    doActionAndWatchConfig(action_data, newValue);
    setValue(newValue);
  }, [checkMinMax, value, doActionAndWatchConfig, action_data]);

  const doActionDown = useCallback(async () => {
    const newValue = checkMinMax(value - 1);
    doActionAndWatchConfig(action_data, newValue);
    setValue(newValue);
  }, [checkMinMax, value, doActionAndWatchConfig, action_data]);

  const checkMinMax = useCallback(
    (tempValue) => {
      if (tempValue > max_value) {
        tempValue = max_value;
      }
      if (tempValue < min_value) {
        tempValue = min_value;
      }
      return tempValue;
    },
    [max_value, min_value]
  );

  return (
    <View style={styles.wrap}>
      <TouchableOpacity style={styles.downButton} onPress={doActionDown}>
        <IconOutline name="down" size={32} color={Colors.Primary} />
      </TouchableOpacity>

      <Text type="H2">{text_format.replace('{number}', value)}</Text>

      <TouchableOpacity style={styles.upButton} onPress={doActionUp}>
        <IconOutline name="up" size={32} color={Colors.Primary} />
      </TouchableOpacity>
    </View>
  );
};

export default NumberUpDownActionTemplate;
