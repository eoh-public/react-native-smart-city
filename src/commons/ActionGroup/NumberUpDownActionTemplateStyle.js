import { StyleSheet } from 'react-native';
import { Colors } from '../../configs';

export default StyleSheet.create({
  wrap: {
    paddingVertical: 25,
    marginHorizontal: 20,
    marginTop: 20,
    borderWidth: 1,
    borderColor: Colors.Gray4,
    borderRadius: 10,
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  downButton: {
    marginRight: 60,
  },
  upButton: {
    marginLeft: 60,
  },
  title: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    color: Colors.Gray7,
    marginTop: 0,
    marginBottom: 10,
  },
  controlPannel: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
  },
});
