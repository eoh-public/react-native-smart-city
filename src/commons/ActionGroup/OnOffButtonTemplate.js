import React, { useCallback, useEffect, useState } from 'react';
import { TouchableOpacity, View } from 'react-native';
import { Icon } from '@ant-design/react-native';
import Text from '../Text';

import { Colors } from '../../configs';
import { watchMultiConfigs } from '../../iot/Monitor';
import { useConfigGlobalState } from '../../iot/states';
import { bigButtonStyles } from './styles';

const OnOffButtonTemplate = ({ actionGroup, doAction }) => {
  const { configuration } = actionGroup;
  const [isOn, setIsOn] = useState(true);
  // eslint-disable-next-line no-unused-vars
  const [configValues, _] = useConfigGlobalState('configValues');
  const triggerAction = useCallback(
    (value) => {
      if (isOn) {
        doAction(configuration.action_off_data);
      } else {
        doAction(configuration.action_on_data);
      }
      configuration.config && watchMultiConfigs([configuration.config]);
    },
    [configuration, doAction, isOn]
  );

  useEffect(() => {
    const configValue = configValues[configuration.config];
    setIsOn(configValue);
  }, [configuration.config, configValues]);

  useEffect(() => {
    watchMultiConfigs([configuration.config]);
  }, [configuration.config]);

  return (
    <>
      <View style={bigButtonStyles.barrierControlContainer}>
        <TouchableOpacity
          style={bigButtonStyles.bigCircle}
          onPress={triggerAction}
        >
          <View style={bigButtonStyles.smallCircle}>
            <Icon
              name={isOn ? configuration.icon_on : configuration.icon_off}
              size={44}
              color={isOn ? Colors.Green7 : Colors.Gray6}
            />
            <Text
              style={[
                bigButtonStyles.textBig,
                { color: isOn ? Colors.Gray8 : Colors.Gray6 },
              ]}
            >
              {isOn ? configuration.text_on : configuration.text_off}
            </Text>
          </View>
        </TouchableOpacity>
      </View>
    </>
  );
};

export default OnOffButtonTemplate;
