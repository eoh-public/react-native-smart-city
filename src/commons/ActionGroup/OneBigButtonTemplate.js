import React from 'react';
import { TouchableOpacity, View } from 'react-native';
import { Icon } from '@ant-design/react-native';
import Text from '../Text';

import { Colors } from '../../configs';
import { bigButtonStyles } from './styles';

const OneBigButtonTemplate = ({ actionGroup, doAction }) => {
  const { configuration } = actionGroup;

  return (
    <>
      <View style={bigButtonStyles.barrierControlContainer}>
        <TouchableOpacity
          style={bigButtonStyles.bigCircle}
          onPress={() => doAction(configuration.action_data)}
        >
          <View style={bigButtonStyles.smallCircle}>
            <Icon name={configuration.icon} size={44} color={Colors.Green7} />
            <Text style={bigButtonStyles.textBig}>{configuration.text}</Text>
          </View>
        </TouchableOpacity>
      </View>
    </>
  );
};

export default OneBigButtonTemplate;
