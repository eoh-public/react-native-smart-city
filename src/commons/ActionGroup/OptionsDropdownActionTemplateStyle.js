import { StyleSheet } from 'react-native';
import { Colors } from '../../configs';

export default StyleSheet.create({
  wrap: {
    paddingVertical: 36,
    paddingHorizontal: 16,
    marginHorizontal: 20,
    marginTop: 20,
    borderWidth: 1,
    borderColor: Colors.Gray4,
    borderRadius: 10,
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  iconAndText: {
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center',
  },
  iconAndTextOption: {
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'flex-end',
  },
  marginRight: {
    marginRight: 10,
  },
  row: {
    flex: 1,
    paddingVertical: 16,
    borderBottomColor: Colors.Gray4,
    borderBottomWidth: 1,
    marginRight: 24,
    marginLeft: 16,
  },
  rowContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    paddingLeft: 16,
  },
  text: {
    fontSize: 16,
    lineHeight: 24,
    color: Colors.Gray9,
  },
  writeNotAvailable: {
    marginLeft: 16,
  },
  scrollView: {
    height: 350,
  },
});
