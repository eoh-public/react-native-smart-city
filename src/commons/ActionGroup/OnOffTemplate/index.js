import React, { memo, useCallback, useEffect, useMemo, useState } from 'react';

import { watchMultiConfigs } from '../../../iot/Monitor';
import { useConfigGlobalState } from '../../../iot/states';
import OnOffButtonTemplate from './OnOffButtonTemplate';
import OnOffSimpleTemplate from './OnOffSimpleTemplate';

const getComponent = (template) => {
  switch (template) {
    case 'on_off_button_action_template': // todo refactor later with backend
      return OnOffButtonTemplate;
    case 'OnOffSimpleActionTemplate':
      return OnOffSimpleTemplate;
    default:
      return () => <></>;
  }
};

const OnOffTemplate = memo(({ actionGroup, doAction, sensor }) => {
  const { configuration } = actionGroup;
  const [isOn, setIsOn] = useState(true);
  // eslint-disable-next-line no-unused-vars
  const [configValues, _] = useConfigGlobalState('configValues');

  const triggerAction = useCallback(() => {
    if (isOn) {
      doAction(configuration.action_off_data);
    } else {
      doAction(configuration.action_on_data);
    }
    if (sensor.is_managed_by_backend) {
      configuration.config && watchMultiConfigs([configuration.config]);
    }
  }, [configuration, doAction, isOn, sensor.is_managed_by_backend]);

  useEffect(() => {
    const { is_on_value, config } = configuration;
    const configValue = configValues[config];

    if (is_on_value && is_on_value.length > 0) {
      setIsOn(is_on_value.includes(configValue));
      return;
    }
    setIsOn(configValue);
  }, [
    configuration.config,
    configValues,
    configuration.is_on_value,
    configuration,
  ]);

  useEffect(() => {
    if (sensor.is_managed_by_backend) {
      watchMultiConfigs([configuration.config]);
    }
  }, [sensor.is_managed_by_backend, configuration.config]);

  const Component = useMemo(() => {
    return getComponent(actionGroup.template);
  }, [actionGroup]);

  return (
    <>
      <Component
        isOn={isOn}
        triggerAction={triggerAction}
        actionGroup={actionGroup}
      />
    </>
  );
});

export default OnOffTemplate;
