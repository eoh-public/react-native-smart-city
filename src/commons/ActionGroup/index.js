import React from 'react';

import OneBigButtonTemplate from './OneBigButtonTemplate';
import OnOffButtonTemplate from './OnOffButtonTemplate';
import ThreeButtonTemplate from './ThreeButtonTemplate';

const ActionGroup = ({ actionGroup, doAction }) => {
  switch (actionGroup.template) {
    case 'three_button_action_template':
      return (
        <ThreeButtonTemplate actionGroup={actionGroup} doAction={doAction} />
      );
    case 'on_off_button_action_template':
      return (
        <OnOffButtonTemplate actionGroup={actionGroup} doAction={doAction} />
      );
    case 'one_button_action_template':
      return (
        <OneBigButtonTemplate actionGroup={actionGroup} doAction={doAction} />
      );
  }
};

export default ActionGroup;
