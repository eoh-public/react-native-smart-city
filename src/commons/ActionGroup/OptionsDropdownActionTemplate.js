import React, { useCallback, useEffect, useMemo, useState } from 'react';
import { View, TouchableOpacity } from 'react-native';
import { IconOutline } from '@ant-design/icons-react-native';
import { Colors } from '../../configs';

import Text from '../../commons/Text';
import { AlertAction, RadioCircle } from '../../commons';
import { useDropdownAction } from './hooks/useDropdownAction';
import { useConfigGlobalState } from '../../iot/states';
import styles from './OptionsDropdownActionTemplateStyle';
import { watchMultiConfigs } from '../../iot/Monitor';
import { TESTID } from '../../configs/Constants';
import IconComponent from '../../commons/IconComponent';
import { t } from 'i18n-js';

function getOptionValue(option) {
  if (option.value_text) {
    return option.value_text;
  }
  return option.value_int;
}

const OptionsDropdownActionTemplate = ({ actionGroup, doAction, sensor }) => {
  const { configuration, title } = actionGroup;
  const { action_data, options, icon, icon_kit_data } = configuration;
  // eslint-disable-next-line no-unused-vars
  const [configValues, setConfigValues] = useConfigGlobalState('configValues');

  const selectedOption = useMemo(() => {
    const currentValue = configValues[configuration.config];
    if (currentValue === null || currentValue === undefined) {
      return configuration.options[0];
    }

    for (let i = 0; i < configuration.options.length; i++) {
      const option = configuration.options[i];
      const value = getOptionValue(option);
      if (value === currentValue) {
        return option;
      }
    }
    return configuration.options[0];
  }, [configValues, configuration.config, configuration.options]);

  const { stateAlert, hideAlertAction, onShowAlert } = useDropdownAction();
  const [selectedIndex, setSelectedIndex] = useState(
    configuration.options.indexOf(selectedOption)
  );

  useEffect(() => {
    setSelectedIndex(configuration.options.indexOf(selectedOption));
  }, [configuration.options, selectedOption]);

  const onDone = useCallback(() => {
    const newOption = options[selectedIndex];
    const value = getOptionValue(newOption);
    doAction(action_data, value);
    if (sensor.is_managed_by_backend) {
      configuration.config && watchMultiConfigs([configuration.config]);
    }
    hideAlertAction();
  }, [
    action_data,
    configuration.config,
    doAction,
    hideAlertAction,
    options,
    selectedIndex,
    sensor.is_managed_by_backend,
  ]);

  const iconKit = !!icon_kit_data && icon_kit_data.icon;

  const displaySelected = useMemo(() => {
    if (!configuration.config) {
      return t('not_available');
    }
    return selectedOption.text;
  }, [configuration.config, selectedOption]);

  return (
    <View style={styles.wrap}>
      <View style={styles.iconAndText}>
        <IconComponent
          icon={icon}
          iconKit={iconKit}
          isSendingCommand={false}
          size={17}
          style={styles.marginRight}
        />
        <Text type="H4">{title}</Text>
      </View>
      <TouchableOpacity style={styles.iconAndTextOption} onPress={onShowAlert}>
        <Text
          testID={TESTID.OPTIONS_DROPDOWN_ACTION_DISPLAY_SELECTED}
          type="Body"
          color={
            displaySelected === t('not_available') ? Colors.Gray6 : Colors.Gray8
          }
          style={styles.marginRight}
        >
          {displaySelected}
        </Text>
        <IconOutline name="right" size={20} />
      </TouchableOpacity>
      <AlertAction
        visible={stateAlert.visible}
        hideModal={hideAlertAction}
        title={stateAlert.title}
        message={stateAlert.message}
        leftButtonTitle={stateAlert.leftButton}
        leftButtonClick={hideAlertAction}
        rightButtonTitle={stateAlert.rightButton}
        rightButtonClick={onDone}
      >
        <View style={styles.wrapRename}>
          {!action_data && (
            <Text
              color={Colors.Gray6}
              style={styles.writeNotAvailable}
              testID={TESTID.OPTIONS_DROPDOWN_ACTION_WRITE_NOT_AVAILABLE}
            >
              {t('write_not_available')}
            </Text>
          )}
          {options.map((item, index) => (
            <View key={index} style={styles.rowContainer}>
              <RadioCircle active={selectedIndex === index} />
              <TouchableOpacity
                testID={TESTID.OPTIONS_DROPDOWN_ACTION_CHOOSING_ITEM}
                style={styles.row}
                onPress={() => {
                  setSelectedIndex(index);
                }}
              >
                <Text style={styles.text}>{item.text}</Text>
              </TouchableOpacity>
            </View>
          ))}
        </View>
      </AlertAction>
    </View>
  );
};

export default OptionsDropdownActionTemplate;
