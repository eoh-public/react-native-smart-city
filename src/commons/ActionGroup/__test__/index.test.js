import React from 'react';
import { TouchableOpacity } from 'react-native';
import renderer, { act } from 'react-test-renderer';

import ActionGroup from '..';

describe('Test ActionGroup', () => {
  const action_data = {
    color: '#00979D',
    command_prefer_over_bluetooth: true,
    command_prefer_over_googlehome: false,
    command_prefer_over_internet: false,
    googlehome_actions: [],
    icon: 'caret-up',
    id: 20,
    key: '5ed1d4dc-a905-47cd-b0c9-f979644bd21a',
  };

  let wrapper;

  test('render ActionGroup three_button_action_template', () => {
    const mockDoAction = jest.fn();
    const actionGroup = {
      template: 'three_button_action_template',
      configuration: {
        action1: '2b949045-8e03-4c07-a855-7794ade2e69c',
        action1_data: {
          color: '#00979D',
          command_prefer_over_bluetooth: true,
          command_prefer_over_googlehome: false,
          command_prefer_over_internet: false,
          googlehome_actions: [],
          icon: 'caret-up',
          id: 9,
          key: '2b949045-8e03-4c07-a855-7794ade2e69c',
        },
        action2: '38347d5e-4418-4ab0-978c-c82f4c034897',
        action2_data: {
          color: '#00979D',
          command_prefer_over_bluetooth: true,
          command_prefer_over_googlehome: false,
          command_prefer_over_internet: false,
          googlehome_actions: [],
          icon: 'stop',
          id: 11,
          key: '38347d5e-4418-4ab0-978c-c82f4c034897',
        },
        action3: 'a492e08c-8cb1-44ee-8ea0-46aaca4e5141',
        action3_data: {
          color: '#00979D',
          command_prefer_over_bluetooth: true,
          command_prefer_over_googlehome: false,
          command_prefer_over_internet: false,
          googlehome_actions: [],
          icon: 'caret-down',
          id: 10,
          key: 'a492e08c-8cb1-44ee-8ea0-46aaca4e5141',
        },
        icon: 'caret-up',
        text: 'UP',
      },
    };
    act(() => {
      wrapper = renderer.create(
        <ActionGroup actionGroup={actionGroup} doAction={mockDoAction} />
      );
    });
    const instance = wrapper.root;
    const buttons = instance.findAllByType(TouchableOpacity);
    expect(buttons.length).toEqual(3);
    act(() => {
      buttons[0].props.onPress();
    });
    expect(mockDoAction).toHaveBeenCalledTimes(1);
    expect(mockDoAction).toHaveBeenCalledWith(
      actionGroup.configuration.action1_data
    );
    act(() => {
      buttons[1].props.onPress();
    });
    expect(mockDoAction).toHaveBeenCalledTimes(2);
    expect(mockDoAction).toHaveBeenCalledWith(
      actionGroup.configuration.action2_data
    );
    act(() => {
      buttons[2].props.onPress();
    });
    expect(mockDoAction).toHaveBeenCalledTimes(3);
    expect(mockDoAction).toHaveBeenCalledWith(
      actionGroup.configuration.action3_data
    );
  });
  test('render ActionGroup one_button_action_template', () => {
    const mockDoAction = jest.fn();
    const actionGroup = {
      template: 'one_button_action_template',
      configuration: {
        action: '5ed1d4dc-a905-47cd-b0c9-f979644bd21a',
        action_data: action_data,
        icon: 'caret-up',
        text: 'UP',
      },
    };
    act(() => {
      wrapper = renderer.create(
        <ActionGroup actionGroup={actionGroup} doAction={mockDoAction} />
      );
    });
    const instance = wrapper.root;
    const buttons = instance.findAllByType(TouchableOpacity);
    expect(buttons.length).toEqual(1);
    act(() => {
      buttons[0].props.onPress();
    });
    expect(mockDoAction).toHaveBeenCalledTimes(1);
    expect(mockDoAction).toHaveBeenCalledWith(action_data);
  });

  test('render ActionGroup on_off_button_action_template', () => {
    const mockDoAction = jest.fn();
    const actionGroup = {
      template: 'on_off_button_action_template',
      configuration: {
        action_on: '5ed1d4dc-a905-47cd-b0c9-f979644bd21a',
        icon_on: 'caret-up',
        text_on: 'ON',
        action_off: '5ed1d4dc-a905-47cd-b0c9-f979644bd21b',
        icon_off: 'caret-up',
        text_off: 'OFF',
      },
    };
    act(() => {
      wrapper = renderer.create(
        <ActionGroup actionGroup={actionGroup} doAction={mockDoAction} />
      );
    });
    const instance = wrapper.root;
    const buttons = instance.findAllByType(TouchableOpacity);
    expect(buttons.length).toEqual(1);
    act(() => {
      buttons[0].props.onPress();
    });
    expect(mockDoAction).toHaveBeenCalledTimes(1);
  });
});
