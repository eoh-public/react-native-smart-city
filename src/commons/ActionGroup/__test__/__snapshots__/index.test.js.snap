// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ActionGroup render ActionGroup one_button_action_template 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "marginTop": 103,
      "padding": 20,
      "width": "100%",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FAFAFA",
        "borderRadius": 200,
        "elevation": 5,
        "height": 237,
        "justifyContent": "center",
        "opacity": 1,
        "shadowColor": "#3031331a",
        "shadowOffset": Object {
          "height": 8,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 12,
        "width": 237,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 200,
          "elevation": 2,
          "height": 195,
          "justifyContent": "center",
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 4,
            "width": 0,
          },
          "shadowOpacity": 0.15,
          "shadowRadius": 12,
          "width": 195,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#00979D",
              "fontFamily": "antoutline",
              "fontSize": 44,
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            undefined,
          ]
        }
      >
        î©€
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "SFProDisplay-Regular",
              "fontSize": undefined,
              "lineHeight": undefined,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {
              "color": "#595959",
              "fontSize": 16,
              "marginTop": 18,
            },
          ]
        }
      >
        UP
      </Text>
    </View>
  </View>
</View>
`;

exports[`Test ActionGroup render ActionGroup three_button_action_template 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderColor": "#E8E8E8",
      "borderRadius": 10,
      "borderWidth": 1,
      "elevation": 3,
      "flexDirection": "row",
      "justifyContent": "space-around",
      "marginHorizontal": 20,
      "marginTop": 20,
      "paddingHorizontal": 8,
      "paddingVertical": 16,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.25,
      "shadowRadius": 3.84,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "aspectRatio": 1,
        "backgroundColor": "#FFFFFF",
        "borderColor": "#E8E8E8",
        "borderRadius": 10,
        "borderWidth": 1,
        "elevation": 3,
        "flex": 1,
        "justifyContent": "center",
        "marginHorizontal": 4,
        "opacity": 1,
        "shadowColor": "#303133",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.25,
        "shadowRadius": 3.84,
      }
    }
  >
    <View
      style={
        Object {
          "height": 28,
          "marginBottom": 7,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#D9D9D9",
              "fontFamily": "antoutline",
              "fontSize": 30,
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            undefined,
          ]
        }
      >
        
      </Text>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "SFProDisplay-Regular",
            "fontSize": undefined,
            "lineHeight": undefined,
            "textAlign": "left",
            "textDecorationLine": "none",
          },
          Object {
            "fontSize": 14,
            "lineHeight": 22,
          },
        ]
      }
    />
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "aspectRatio": 1,
        "backgroundColor": "#FFFFFF",
        "borderColor": "#E8E8E8",
        "borderRadius": 10,
        "borderWidth": 1,
        "elevation": 3,
        "flex": 1,
        "justifyContent": "center",
        "marginHorizontal": 4,
        "opacity": 1,
        "shadowColor": "#303133",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.25,
        "shadowRadius": 3.84,
      }
    }
  >
    <View
      style={
        Object {
          "height": 28,
          "marginBottom": 7,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#D9D9D9",
              "fontFamily": "antoutline",
              "fontSize": 30,
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            undefined,
          ]
        }
      >
        
      </Text>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "SFProDisplay-Regular",
            "fontSize": undefined,
            "lineHeight": undefined,
            "textAlign": "left",
            "textDecorationLine": "none",
          },
          Object {
            "fontSize": 14,
            "lineHeight": 22,
          },
        ]
      }
    />
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "aspectRatio": 1,
        "backgroundColor": "#FFFFFF",
        "borderColor": "#E8E8E8",
        "borderRadius": 10,
        "borderWidth": 1,
        "elevation": 3,
        "flex": 1,
        "justifyContent": "center",
        "marginHorizontal": 4,
        "opacity": 1,
        "shadowColor": "#303133",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.25,
        "shadowRadius": 3.84,
      }
    }
  >
    <View
      style={
        Object {
          "height": 28,
          "marginBottom": 7,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#D9D9D9",
              "fontFamily": "antoutline",
              "fontSize": 30,
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            undefined,
          ]
        }
      >
        
      </Text>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "SFProDisplay-Regular",
            "fontSize": undefined,
            "lineHeight": undefined,
            "textAlign": "left",
            "textDecorationLine": "none",
          },
          Object {
            "fontSize": 14,
            "lineHeight": 22,
          },
        ]
      }
    />
  </View>
</View>
`;
